libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_network_ble.c

```diff
@@ -76,6 +84,34 @@ static void _callback( IotBleDataTransferChannelEvent_t event,
             pBleConnection->pCallback( pBleConnection, pBleConnection->pContext );
             break;
 
+        case IOT_BLE_DATA_TRANSFER_CHANNEL_ERROR_UPDATED:
+
+            /* Currently only used for logging, but could short-circuit failure here.
+             * At time of writing, such error events could occur but operations would only fail LATER,
+             * after timeout that resulted from this error */
+            ucPeerError = ( BleNetworkError_t ) IotBleDataTransfer_GetPeerError( pChannel );
```

Comment:
By making ucPeerError part of _bleNetworkConnection_t,  the `IotNetworkBle_Create()` code can check for the errors and log them appropriately by not failing with a timeout.

Reply:
**Summary:** 
Due to interdependencies, covering all short-circuiting failures requires code changes at higher and lower level APIs with respect to `iot_network_ble.c`. Deferring short-circuiting to later PR.

**Details**
I have code that will short circuit fail IotNetworkBle_Create() if a `BLE_PEER_COGNITO_AUTH_FAIL` was received, but it doesn't cover all short-circuitable errors cases. 

While the behavior of `IotNetworkBle_Create()` is largely defined at the `iot_network_ble.c` level, thereby limiting required short-circuiting code changes to `iot_network_ble.c`, the `IotNetworkBle_Send/Receive()` calls are mostly defined by `IotBleDataTransfer_Send` from lower-level API `iot_ble_data_transfer.h`. At this lower level the error codes, similar to the tx/rx data, are arbitrary values that are handled by the higher level implementation like `iot_network_ble.c`. By itself it cannot determine whether a received error should short-circuit and cutoff a connection. 

Moreover, the callers _above_ `iot_network_ble.c` that use `IotNetworkBle_Send/Receive` have their own behavior and receive callbacks which likely need come into play too.